src/java/boa/aggregators/ml/MLAggregator.java

```diff
@@ -0,0 +1,317 @@
+            } catch (final Exception e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    public void saveModel(Object model) {
+        FSDataOutputStream out = null;
+        FileSystem fileSystem = null;
+        Path filePath = null;
+        ObjectOutputStream objectOut = null;
+        try {
+            JobContext context = (JobContext) getContext();
+            Configuration configuration = context.getConfiguration();
+            int boaJobId = configuration.getInt("boa.hadoop.jobid", 0);
+            JobConf job = new JobConf(configuration);
+            Path outputPath = FileOutputFormat.getOutputPath(job);
+            fileSystem = outputPath.getFileSystem(context.getConfiguration());
+
+            if (DefaultProperties.localOutput != null) {
+                fileSystem.mkdirs(new Path(DefaultProperties.localOutput, new Path("" + boaJobId)));
+                filePath = new Path(DefaultProperties.localOutput, new Path("" + boaJobId, new Path(("" + getKey()).split("\\[")[0] + System.currentTimeMillis() + "ML.model")));
+            } else {
+                fileSystem.mkdirs(new Path(DefaultProperties.HADOOP_OUT_LOCATION, new Path("" + boaJobId)));
+                filePath = new Path(DefaultProperties.HADOOP_OUT_LOCATION, new Path("" + boaJobId, new Path(("" + getKey()).split("\\[")[0] + System.currentTimeMillis() + "ML.model")));
+            }
+
+
+            if (fileSystem.exists(filePath))
+                return;
+
+            out = fileSystem.create(filePath);
+
+            ByteArrayOutputStream byteOutStream = new ByteArrayOutputStream();
+            objectOut = new ObjectOutputStream(byteOutStream);
+            objectOut.writeObject(model);
+            byte[] serializedObject = byteOutStream.toByteArray();
+
+            out.write(serializedObject);
+
+            this.collect(filePath.toString());
+
+        } catch (Exception e) {
+            e.printStackTrace();
+        } finally {
+            try {
+                if (out != null) out.close();
+                if (objectOut != null) objectOut.close();
+            } catch (final Exception e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    protected void applyFilterToUnfilteredInstances(Filter filter) throws Exception {
+        unFilteredInstances = Filter.useFilter(unFilteredInstances, filter);
+    }
+
+    protected void applyFilterToUnfilteredInstances(Filter filter, Instances filteredInstances) throws Exception {
+        unFilteredInstances = Filter.useFilter(unFilteredInstances, filter);
+        moveFromUnFilteredToFiltered(filteredInstances);
+    }
+
+    protected void moveFromUnFilteredToFiltered(Instances filteredInstances) {
+        int totalUnfilteredInstances = unFilteredInstances.numInstances();
+        filteredInstances.addAll(unFilteredInstances.subList(0, totalUnfilteredInstances));
+        while (totalUnfilteredInstances-- > 0) {
```

Comment:
replace loop with `unFilteredInstances.delete()`

Reply:
I have fixed it.